hecuba_core/src/ArrayDataStore.h

```diff
@@ -18,24 +18,26 @@ class ArrayDataStore {
 
     ~ArrayDataStore();
 
-    void store(const uint64_t *storage_id, ArrayMetadata* metadata, void *data) const;
-
+    void store(const uint64_t *storage_id, ArrayMetadata *metadata, void *data) const;
+    void store_numpy_to_cas(const uint64_t *storage_id, ArrayMetadata *metadata, void *data, std::list<std::vector<uint32_t> > coord) const;
     void *read(const uint64_t *storage_id, ArrayMetadata *metadata) const;
 
     // Returns the metadata of the array identified by the storage_id
     ArrayMetadata *read_metadata(const uint64_t *storage_id) const;
 
     // Overwrite the metadata of the array identified by the given storage_id
-    void update_metadata(const uint64_t *storage_id, ArrayMetadata* metadata) const;
+    void update_metadata(const uint64_t *storage_id, ArrayMetadata *metadata) const;
 
+    void *read_n_coord(const uint64_t *storage_id, ArrayMetadata *metadata, std::list<std::vector<uint32_t> > crd,
+                       void *save);
 
-private:
 
+private:
 
 
     CacheTable *cache = nullptr, *read_cache = nullptr;
 
-    SpaceFillingCurve partitioner;
+    SpaceFillingCurve *partitioner;
```

Comment:
Why a pointer, and if you need a pointer, remember to delete the object inside the destructor (it is not destroyed, and you removed the destruction of the cache and read_cache too)